backend/src/main/java/com/summerHack/diningTogether/repository/FoodRepositoryImpl.java

```diff
@@ -33,6 +40,19 @@ public List<Food> findByParameters(
         confirmed.ifPresent(c -> predicates.add(cb.equal(food.get("completed"), c)));
 
         query.where(predicates.toArray(Predicate[]::new));
-        return em.createQuery(query).getResultList();
+        List<Food> result = em.createQuery(query).getResultList();
+        if (distance.isPresent() == false)
+            return result;
+        return result.parallelStream()
```

Comment:
This can be super inefficient because the time complexity is O(n) where n is all locations stored in the database. You can try use the geographic functionality in redis or just keep it for convenience.

Reply:
Well, I used parallelStream which means I count the distance once